lib/awful/widget/calendar.lua

```diff
@@ -0,0 +1,360 @@
+-- @treturn wibox The wibox calendar
+function calendar_popup:attach(widget, position)
+    local position = position or "tr"
+    widget:buttons(util.table.join(abutton({ }, 1, function ()
+                                                              self:call_calendar(0, {position=position})
+                                                              self.visible = not self.visible
+                                                        end),
+                                   abutton({ }, 4, function () self:call_calendar(-1) end),
+                                   abutton({ }, 5, function () self:call_calendar( 1) end)))
+    return self
+end
+
+
+--- Return a new calendar wibox by type.
+--
+-- A calendar widget displaying a `month` or a `year`
+-- @tparam string caltype Type of calendar `month` or `year`
+-- @tparam table args Properties of the widget
+-- @treturn wibox A wibox containing the calendar
+local function get_cal_wibox(caltype, args)
+    if caltype ~= "month" and caltype ~= "year" then return end
+    local args = args or {}
+
+    local ret = wibox{ ontop   = true,
+                       opacity = args.opacity or 1,
+                       bg      = args.bg
+    }
+    util.table.crush(ret, calendar_popup, false)
+
+    ret.offset   = 0
+    ret.position = args.position  or "cc"
+    ret.screen   = args.screen
+
+    local widget = wibox.widget {
+        font         = args.font,
+        spacing      = args.spacing,
+        week_numbers = args.week_numbers,
+        start_sunday = args.start_sunday,
+        fn_embed     = embed(parse_all_options(args)),
+        widget = wibox.widget.calendar(caltype)
+    }
+    ret:set_widget(widget)
+
+    ret:buttons(util.table.join(
+            abutton({ }, 1, function () ret.visible=false end),
+            abutton({ }, 3, function () ret.visible=false end),
+            abutton({ }, 4, function () ret:call_calendar(-1) end),
+            abutton({ }, 5, function () ret:call_calendar( 1) end)
+    ))
+    return ret
+end
+
+--- A month calendar wibox.
+--
+-- It is highly customizable using the same options as for the widgets.
+-- The options are set once and for all at creation, though.
+--
+--@DOC_wibox_awidget_calendar_month_wibox_EXAMPLE@
+--
+--    local mytextclock = wibox.widget.textclock()
+--    month_calendar:attach( mytextclock, "tr" )
+--
+-- @tparam table args Properties of the widget
+-- @treturn wibox A wibox containing the calendar
+-- @function awful.calendar.month
+function calendar_popup.month(args)
+    return get_cal_wibox("month", args)
+end
+
+
+--- A year calendar wibox.
+--
+-- It is highly customizable using the same options as for the widgets.
+-- The options are set once and for all at creation, though.
+--
+--@DOC_wibox_awidget_calendar_year_wibox_EXAMPLE@
+--
+--    globalkeys = awful.util.table.join(globalkeys, awful.key({ modkey, "Control" }, "c",  function () year_calendar:toggle() end))
```

Comment:
Change `awful.util.table.join` to `gears.table.join`

Also luacheck line too long (132 > 120)